src/color_gradients.jl

```diff
@@ -43,7 +43,11 @@ end
 
 getindex(cl::ColorLibrary, key::Symbol) = getgradient(key, cl)
 
-function register_gradient_colors(name::Symbol, colors::AbstractVector{C}, color_library::Symbol = :default) where {C<:Colorant}
+const _categorical_gradients = Symbol[]
+is_categorical(grad) = grad in _categorical_gradients
```

Comment:
I see that you've made categoricalness a trait here rather than making a type for it and using dispatch - what's your thinking here?

Reply:
The way `ColorLibrary` is implemented now it just stores names and corresponding Vectors of Colors. If I call `plot(..., palette = name)` where `name` is a `Symbol` how can we know if we should use the current bahaviour of finding the most different colors in a gradient defined by `name` or just using the vector of colors stored under `name` as palette. As I wrote in my other comment this would require updating `ColorLibrary` to store different types for `ColorGradient` and `ColorPalette`.